packages/shared/src/pipelines.ts

```diff
@@ -0,0 +1,95 @@
+export interface Operation<T, D = unknown> {
+  invoke(data: D): T;
+}
+
+export class Pipeline<T, D> implements Operation<T[]> {
+  private operations: Operation<T>[] = [];
+
+  addOperation(operation: Operation<T>) {
+    this.operations.push(operation);
+    return this;
+  }
+
+  invoke(data: D): T[] {
+    return this.operations.map(operation => operation.invoke(data));
+  }
+}
+
+export interface IOOperation<T, D = unknown, O = unknown> {
+  get(): T;
+  set(data: D, options: O): void;
+  remove(): void;
+}
```

Comment:
(This looks like the missing interface I was talking about in #3362)

This feels a little too complicated to me because it looks like we're defining 4 interfaces (and some duplicates across this PR and #3362 ) to simply invoke 2 methods. However, I want to think more about it as I'm not familiar with the pattern.

Could we achieve the same results by defining a method/helper with the following signature?

```
function writeCookie(handlers: CookieHandler[]) {
  handlers.forEach(h => h.write(...))
}
```

or similar? Would that be simpler?

Reply:
This PR handles both generating multiple values based on a single input (case of backend SDK that uses a single cookie directive and produces the suffixed and un-suffixed variants) and performing multiple actions triggering a single pipeline method (eg write both suffixed and un-suffixed cookies using a single cookie key and value).
That's why there are multiple interfaces that look similar.
To achieve the same using method / helpers we should:
- write 3 cookie handlers (eg session, clientUat, devBrowser) for ClerkJS
- write 3 functions to generate the cookie key variants (eg session, clientUat, devBrowser)  for backend SDK

For me this seems like the same amount of class or methods but different implementation. Using a pattern (even though it would require some time to learn it) would make the handling of all those cases consistent and expected instead of trying to read and verify what it's separate handler or function does.
If you think that it's more complex than it should, let's drop the pattern usage and simplify things. @nikosdouvlis Feel free to provide any suggestion on how we could make things simpler?